web-ui/src/main/resources/catalog/components/search/map/MapFieldDirective.js

```diff
@@ -134,20 +134,253 @@
               var updateField = function(geom) {
                 feature.setGeometry(geom);
 
-                // Write the extent as 4326 WKT polygon
-                var lonlatFeat, writer, wkt;
+                // Get projection code of current map view
+                var proj = scope.map.getView().getProjection().getCode();
+
+                /**
+                 * Transforms the given longitude-latitude pair to XY coordinates in the current projection.
+                 * @param {number} lon 
+                 * @param {number} lat 
+                 */
+                var getXY = function(lon, lat) {
+                  return (new ol.geom.Point([lon, lat])).transform('EPSG:4326', proj).getFirstCoordinate();
+                }
+
+                // Define polar coordinates in current projection
+                var southPole = getXY(0, -90);
+                var northPole = getXY(0, 90);
+
+                /**
+                 * Checks if the feature geometry contains the north or south pole.
+                 * Returns the polar coordinate in the current map projection when found or null otherwise.
+                 * NOTE: Will not trigger if pole is on the edge.
+                 * @param {ol.feature} feat
+                 */
+                var getInsidePole = function(feat) {
+                  let geom = feat.getGeometry();
+                  if (geom.intersectsCoordinate(southPole)) {
+                    return southPole;
+                  } else if (geom.intersectsCoordinate(northPole)) {
+                    return northPole;
+                  }
+                }
+
+                /**
+                 * Adds points to line segments.
+                 * Specify the amount of segment parts to create (default = 4).
+                 * The number of points that will be added is (parts - 1).
+                 * @param {ol.feature} feat
+                 * @param {int} parts
+                 */
+                var densifyEdges = function(feat, parts=4) {
+                  let outCoords = [];
+                  let inCoords = feat.getGeometry().getCoordinates()[0];
```

Comment:
This will not work for linestrings right? maybe this should be documented better, the header only mentions "line segments"

Reply:
For lines, this issue does not apply. Lines will reproject just fine.
But does the user even have the possibility to draw a line for a search query? Does that even make sense? People are interested in areas usually?

Reply:
Oh no, queries will always use polygons, I was just suggesting to make that clearer in the function documentation :)

Reply:
Ok, will change!